src/large_pages/node_large_page.cc

```diff
@@ -212,6 +214,41 @@ static struct text_region FindNodeTextRegion() {
     }
     start += cursz;
   }
+#elif defined(__APPLE__)
+  struct text_region nregion;
+  nregion.found_text_region = false;
+  struct vm_region_submap_info_64 map;
+  mach_msg_type_number_t count = VM_REGION_SUBMAP_INFO_COUNT_64;
+  vm_address_t addr = 0UL;
+  vm_size_t size = 0;
+  natural_t depth = 1;
+
+  while (true) {
+    if (vm_region_recurse_64(mach_task_self(), &addr, &size, &depth,
+          (vm_region_info_64_t)&map, &count) != KERN_SUCCESS) {
```

Comment:
`reinterpret_cast` please, no C style casts (and preferably line up the arguments.)